instrumentation/opentelemetry-instrumentation-psycopg2/tests/test_psycopg2_instrumentation.py

```diff
@@ -0,0 +1,183 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# type: ignore
+
+
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from importlib.metadata import PackageNotFoundError
+from unittest import TestCase
+from unittest.mock import call, patch, Mock
+
+from opentelemetry.instrumentation.psycopg2 import Psycopg2Instrumentor
+from opentelemetry.instrumentation.psycopg2.package import (
+    _instruments,
+    _instruments_psycopg2,
+    _instruments_psycopg2_binary,
+)
+from opentelemetry.instrumentation.auto_instrumentation._load import (
+    _load_instrumentors,
+)
+
+
+
+class TestPsycopg2InstrumentationDependencies(TestCase):
```

Comment:
@emdneto, I think i got it here.

I added some tests first to make sure that the instrumentation_dependencies functions handle both, one or none installed.

The last test, test_instruments_with_psycopg2_installed, tests higher up in the auto instrumentation flow to make sure that the individual error around one package working but not the other doesn't happen again.